include/verilated_replay_common.cpp

```diff
@@ -0,0 +1,83 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+//
+// Copyright 2003-2020 by Todd Strader. This program is free software; you can
+// redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License.
+// Version 2.0.
+//
+// Verilator is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+//=========================================================================
+///
+/// \file
+/// \brief Verilator: Common replay code
+///
+///     See verilator_replay
+///
+/// Code available from: http://www.veripool.org/verilator
+///
+//=========================================================================
+
+
+#include "verilated_replay_common.h"
+#include <iostream>
+
+using namespace std;
+
+// TODO -- can we not do this?
+// Include the GTKWave implementation directly
+// Ugh, building with verilated_fst_c.cpp, brings this in, let's really not do this
+//#define FST_CONFIG_INCLUDE "fst_config.h"
+//#include "gtkwave/fastlz.c"
+//#include "gtkwave/fstapi.c"
+//// TODO -- use the system's LZ4 library, not this copy
+//#include "gtkwave/lz4.c"
```

Comment:
This was copied from verilated_fst_c, but then I had to comment it out when I included that code.  Two things I propose doing in separate PRs:
- Compile the FST API as its own object in the Verilator Makefiles
- Have configure check for a system version of LZ4 and use that if it exists

The former will be complicated by the fact that I am not using this code in a compiled Verilator tool in addition to the runtime.

The latter is obviously a problem if the system's LZ4 is not compatible with the FST API.  However, the way we're currently doing this now is a problem if you link the verilated code with something else that uses a different LZ4 (I've been bitten by this).

Let me know if you think these two items would be good improvements and I'll open new issues.

Reply:
To use the system library lz4 you'll need Gtkwave to agree.  In general I suspect this library won't be well available and opens door to strange compatibility problems based on if the file is there or not.

Perhaps an alternative is perhaps to make the functions static?  E.g. add a FST_STATIC define that is optionally set we submit upstream, and we set it.  If agree suggest make a bug/pull on us and on gtkwave.